Add LibFeatures module

gavin-norman-sociomantic opened the pull request:
Meta module defining constants that match swarm feature releases
allowing downstream libraries to conditionally implement
functionality relying on later swarm versions.

nemanja-boric-sociomantic commented:
Is there any point in including this file earlier? (I mean, to go via patch release)

gavin-norman-sociomantic commented:
Ha, we could have released it in v3.3.0 :D

nemanja-boric-sociomantic commented:
Well, I mean - is there any practical point of getting this into a patch release, meaning - will the downstream user benefit from providing this file earlier. If not, I think it's fine to include it in v4.2.0.

mihails-strasuns-sociomantic commented:
Well I think we can be pretty sure that only possible users of this functionality are you yourselves (in protos) so unless there is a use case calling for earlier inclusion now, 4.2.0 is OK

mihails-strasuns-sociomantic commented on src/swarm/LibFeatures.d:
```diff
@@ -0,0 +1,23 @@
+
+module swarm.LibFeatures;
+
+const has_features_4_2 = true;
```
Maybe sanity check test in `LibFeatures_test`? :) i.e. `static assert ( hasFeaturesFrom!("swarm", 4, 0));`

gavin-norman-sociomantic replied:
Updated.

Commit:
Add LibFeatures module

Meta module defining constants that match swarm feature releases
allowing downstream libraries to conditionally implement
functionality relying on later swarm versions.
